src/overlays/actors/ovl_Obj_Demo/z_obj_demo.c

```diff
@@ -29,14 +29,90 @@ const ActorInit Obj_Demo_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjDemo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDemo* this = THIS;
+    s16 temp_v0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Demo/ObjDemo_Init.s")
+    thisx->params &= 0xFF;
+    if ((thisx->params != 0xFF) && (Flags_GetSwitch(globalCtx, thisx->params))) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
```

Comment:
```suggestion
        Actor_MarkForDeath(&this->actor);
        return;
    }
```
One of the few cases where this is preferred.